src/oemof/solph/_options.py

```diff
@@ -144,85 +145,56 @@ class NonConvex:
-        }
-
-        for attribute in set(
-            scalars + sequences + dictionaries + list(kwargs)
-        ):
-            value = kwargs.get(attribute, defaults.get(attribute))
-            if attribute in dictionaries:
-                setattr(
-                    self,
-                    attribute,
-                    {"ub": sequence(value["ub"])},
-                )
-            else:
-                setattr(
-                    self,
-                    attribute,
-                    sequence(value) if attribute in sequences else value,
-                )
+    def __init__(
+        self,
+        initial_status=0,
+        minimum_uptime=0,
+        minimum_downtime=0,
+        maximum_startups=None,
+        maximum_shutdowns=None,
+        startup_costs=None,
+        shutdown_costs=None,
+        activity_costs=None,
+        inactivity_costs=None,
+        negative_gradient_limit=None,
+        positive_gradient_limit=None,
+        custom_attributes=None,
+    ):
+        if custom_attributes is None:
+            custom_attributes = {}
+
+        self.initial_status = initial_status
+        self.minimum_uptime = minimum_uptime
+        self.minimum_downtime = minimum_downtime
+        self.maximum_startups = maximum_startups
+        self.maximum_shutdowns = maximum_shutdowns
+
+        self.startup_costs = sequence(startup_costs)
+        self.shutdown_costs = sequence(shutdown_costs)
+        self.activity_costs = sequence(activity_costs)
+        self.inactivity_costs = sequence(inactivity_costs)
+        self.negative_gradient_limit = sequence(negative_gradient_limit)
+        self.positive_gradient_limit = sequence(positive_gradient_limit)
+
+        for attribute, value in custom_attributes.items():
+            setattr(self, attribute, value)
 
         self._max_up_down = None
 
-    def _calculate_max_up_down(self):
-        """
-        Calculate maximum of up and downtime for direct usage in constraints.
+    @property
+    def max_up_down(self):
+        """Return maximum of minimum_uptime and minimum_downtime.
 
         The maximum of both is used to set the initial status for this
-        number of timesteps within the edge regions.
+        number of time steps within the edge regions.
         """
-        if self.minimum_uptime is not None and self.minimum_downtime is None:
-            max_up_down = self.minimum_uptime
-        elif self.minimum_uptime is None and self.minimum_downtime is not None:
-            max_up_down = self.minimum_downtime
-        else:
-            max_up_down = max(self.minimum_uptime, self.minimum_downtime)
-
-        self._max_up_down = max_up_down
-
-    @property
-    def max_up_down(self):
-        """Compute or return the _max_up_down attribute."""
-        if self._max_up_down is None:
-            self._calculate_max_up_down()
 
-        return self._max_up_down
+        return max(self.minimum_uptime, self.minimum_downtime)
```

Comment:
The check has become obsolete since you are setting the default values to 0, right?
We don't want to allow the user for actively setting either one to None, do we?

Reply:
We still need the longer time of the two. However, I forgot to delete _max_up_down which is no longer used.